### PR TITLE
Added the PML test and changed the FDTD Scheme to implement it  in One Dimension.

### PR DESCRIPTION
**Pull request by sergiorf2001**
We, David González, Juan José Rojas and Sergio Rodríguez (me), have implemented the Perfectly Matched Layers to the program (which is our assigned Final Project of this part of MCFNL). When doing this, we have changed the FDTD Scheme, but when self.conPML=0 (array), it gets back to the previous one. Also implemented a test which checks that after crossing the PML, the energy is equal to zero (under a certain tolerance condition). Also changed the visualization code to make it a bit faster.

### CODE REVIEW

File: fdtd1d.py [hunk 1]
```diff
@@ -57,17 +58,49 @@ def set_conductivity_regions(self, regions):
...
             raise RuntimeError(
                 "Initial condition not set. Call set_initial_condition first.")
 
         self.e_old_left = self.e[1]
         self.e_old_right = self.e[-2]
-
-        self.h[:] = self.h[:] - dt / self.dx / MU0 * (self.e[1:] - self.e[:-1])
+        
+        '''
+        Udatede FDTD Scheme to implement PML and conductivity.
```
Comment (lmdiazangulo) [inline fdtd1d.py:98]: Typo

File: test_fdtd1d.py [hunk 2]
```diff
@@ -229,7 +229,37 @@ def test_fdtd_1d_solver_conductivity():
...
+    solver.set_initial_condition(initial_condition)
+    
+    # Set PML
+    solver.set_PML(thicknessPML, m, R0, dx)
+    
+    # Run the simulation
+    solver.run_until(Tf, dt)
 
+    # Check that the energy after the wave have passed through the PML is almost 0 in the whole grid
+    assert solver.energy[-1]/solver.energy[0] < 0.05
```
Comment (lmdiazangulo) [inline test_fdtd1d.py:262]: Please change this test to assess that the required reflection coefficient from the PML is what is expected.

**Review by lmdiazangulo — CHANGES_REQUESTED** [review]

File: test_fdtd1d.py (hunk 2 above)
Comment (sergiorf2001) [reply test_fdtd1d.py:262]: We were not a able to introduce a test regarding the reflection coefficient yet, so we implemented this one. We made the pull request so we wouldn't have problems later, in case it is updated by another student before we finish it.

File: fdtd1d.py (hunk 1 above)
Comment (sergiorf2001) [reply fdtd1d.py]: "Udadete" typo updated.

**Comment by sergiorf2001** [comment]
We updated the tests adding the reflection coefficient test.

**Review by lmdiazangulo — APPROVED** [review]
